### DISCUSSION TITLE
ButtonSecondary hover

### DISCUSSION THREAD

**Discussion by yceballost** [Telefonica/mistica-design#496] [Proposals]
Desde Movistar España se cuestiona que el hover del botón secondary no es muy evidente. 

Propuesta del equipo de Canal Online
https://xd.adobe.com/view/25839da2-f72d-4f1d-80cd-453bcbf8a635-70d0/

**Comment by yceballost** [comment]
Es cierto que por la anatomía de esta tipología de botones, al tener un borde tan fino los cambios de estado son muy poco apreciables. Creo que la propuesta de COL mejora algo la problemática pero realmente tampoco creo que marque una diferencia increíble con respecto a lo que hay actualmente.

https://ibit.ly/Iwbt

**Reply by aweell** [reply]
Si coincido que con la evolución de Movistar el cambio de estado se vuelve casi imperceptible.

Algo que puede solucionar esta problemática:

https://codepen.io/AlexBueno/pen/rNdmaPE

Sería reutilizando el valor que actualmente tiene el token `buttonLinkBackgroundSelected`.

Propondría además separar el estado `:hover` en un nuevo token para el `buttonSecondary` al igual que se hace actualmente en el primary con `buttonPrimaryBackgroundHover`

**Reply by yceballost** [reply]
habría que ver cómo podría ser el resto de estados (inverse o en dark mode por ejemplo)

qué piensas sobre esta solución @dsys-col @gitbraga? 

(en el codepen de Alex puedes cambiar movistar por vivo y ver como sería en vivo :D )

**Reply by dsys-col** [reply]
Buenos días. 
Poner el bg del hover en un fondo sólido puede ser una solución, aunque no nos apasiona ya que convertimos en el efecto un botón secundario a un primario visualmente, aunque sea por un momento. Pero si no encontramos nada mejor puede ser la solución. He cambiado el color de fondo al movistar Darkblue y parece que funciona con este fondo. Hay que ver cómo funciona con los inversos, no sé si en ese caso es tan buena idea. Solo unos comentarios:

1. **¿Que color es bg del hover?** En el código del codepen aparece este (_--button-link-background-selected_), pero no nos dice mucho. Tendrá que ser un color que tengamos declarado o un color con opacidad. 

2. **El efecto del hover es algo brusco**. Quizás se pueda suavizar un poco ese efecto con una transición ease in-out inicialmente de 0.3 segundos... Probarlo a ver qué os parece. Podeís jugar con los tiempos, pero creo que 0,3 puede estar bien.

  -webkit-transition: all 0.3s ease-in-out;
  -moz-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;

3. Aunque nos preocupa menos el inverso, quizás deberíamos ver cómo queda este efecto...**¿Haríamos lo mismo con el inverso? ¿Podríamos hacer el mismo ejercicio con el inverso?**

**Reply by aweell** [reply]
Buenas @dsys-col,

Aún añadiendo un fondo sólido a la pseudo-clase del button secondary, el contraste del primary default con respecto al fondo (en light mode) es 2.94:1 en comparación al secondary:hover, 1.11:1, por lo que desde mi punto de vista mantiene, perceptualmente, su condición de "primario".



**1.** Los valores para los background del `:hover` son:

Default: `--button-link-background-selected` (`#e6f5fd`)
Inverse: `--button-link-background-selected-inverse` (`hsla(0,0%,100%,.2)`)

Estos valores son los definidos en mística web para la variante button link en sus estados `:hover` y `:active`:

Storybook: https://mistica-n0g2juz14-tuentisre.vercel.app/?path=/story/components-buttons--type-of-buttons
Figma: https://www.figma.com/file/cwUAe7pwaatgrezyLMHRfc/Movistar?node-id=0%3A1

**2.** En las transiciones de color entre estados no me he metido puesto que este ejercicio es únicamente un test de color.

**3.** He actualizado el [codepen](https://codepen.io/AlexBueno/pen/rNdmaPE) para poder ver la versión `inverse` comparativa de los dos botones.

-----

@gitbraga, para ver los estilos en vivo, tendrías que reemplazar el primer import del css por esto:

```
@import "https://cdn.jsdelivr.net/npm/mistica-css@2.3.1/vivo/vivo.css";
```

Para las versiones dark dos opciones:

1. Cambiar el tema de vuestro sistema operativo a dark (automáticamente el codepen detectará el cambio y actualizara los colores de background y botones.

2. Añadir al import -dark al final de la url:

```
@import "https://cdn.jsdelivr.net/npm/mistica-css@2.3.1/vivo/vivo-dark.css";
```
**Nota: hay un error en el :hover dark del primario. Revisaría además @yceballost el bg del button link en dark, pero ese es otro cantar 🙈**

**Reply by dsys-col** [reply]
Creo que no me he debido explicar bien. Me refería a que para que un botón sea percibido como primario o no basicamente se basa en tener un background o no tenerlo, ya que la jerarquía con botones con fondo son percibidos como más pesados. Por tanto, no me estaba refiriendo al contraste, es más un tema de percepción que no tiene que ver con el contraste. El hover lo convierte por así decirlo en un botón con fondo, que es algo que no nos gusta tanto, aunque al producirse en el hover entendemos que tampoco es un problema que el usuario pueda confundir. 

Lo del efecto de transición es por si lo necesitáis tenerlo en cuenta para que no sea muy brusco. Entendemos que esto es una prueba. Si lo vaís a usar, ahí tenéis la aportación.

Muchas gracias por la aclaración, el trabajo y el ejercicio. Por nuestra parte, lo anotamos y aplicaremos estos cambios de mejora en cuanto nos sea posible y cualquier duda lo hablamos.

**Reply by yceballost** [reply]
Empezando por la animación, el botón ya trae esa transición, así que de aplicarlo, probablemente ya tuviese una transición muy parecida a la que propones @dsys-col. Los codepen no son los componentes reales (reales = producción) de mística y esos pequeños detalles no se suelen representar (siendo el punto de la discussion algo diferente a las animaciones, claro)

Personalmente pienso que no habría problema en aplicar un background a un estado hover de un secondary, el botón por defecto se mostrará con el borde y demás diferenciándolo bien del primary, el estado hover es un estado que solo ocurre cuando el usuario posiciona conscientemente el cursor encima del elemento y por tanto es consciente del feedback visual de ese estado hover. Esta solución me parece más óptima que la actual porque el cambio de color a un tono más oscuro sobre el secondary como ya sacó COL es poco evidente.

**Comment by yceballost** [comment]
Se va a trabajar para Q4 para añadirle el background para mejorar el estado hover del button secondary.
https://github.com/Telefonica/mistica-design/issues/697
